jetbrains-core/tst/software/aws/toolkits/jetbrains/services/ecs/exec/RunCommandDialogTest.kt

```diff
@@ -114,36 +93,44 @@ class RunCommandDialogTest {
         val programPath = makeSampleCliExecutable()
         runInEdtAndWait {
             val environmentVariables = mutableListOf<String>()
-            val counter = CountDownLatch(5)
-            val execCommand = RunCommandDialog(projectRule.project, container, connectionSettings)
-                .buildExecCommandConfiguration(('"' + command + '"'), programPath)
             val environment =
                 ExecutionEnvironmentBuilder
                     .create(
                         projectRule.project,
                         DefaultRunExecutor.getRunExecutorInstance(),
-                        RunCommandRunProfile(execCommand, SimpleDataContext.getProjectContext(projectRule.project), connectionSettings)
+                        RunCommandRunProfile(
+                            connectionSettings?.toEnvironmentVariables(),
+                            RunCommandDialog(projectRule.project, container, connectionSettings)
+                                .constructExecCommandParameters(command),
+                            containerName, programPath.toAbsolutePath().toString()
+                        )
                     )
                     .build {
                         it.processHandler?.addProcessListener(object : ProcessAdapter() {
-
                             override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
                                 super.onTextAvailable(event, outputType)
                                 environmentVariables.add(event.text.split("\n").first())
-                                counter.countDown()
+                            }
+                        })
+                        it.processHandler?.addProcessListener(object : ProcessAdapter() {
+                            override fun processTerminated(event: ProcessEvent) {
+                                super.processTerminated(event)
+                                assertThat(environmentVariables[1]).isEqualTo("AccessEcsExecDummy")
```

Comment:
how are these created?

We have helper functions that generate region codes and credentials:

* `software.aws.toolkits.core.region.AwsRegionTestKt#anAwsRegion`
* `software.aws.toolkits.core.credentials.MocksKt#aCredentialsIdentifier`

Rather than hard-coding `us-east-1` into tests which can make certain assumptions that are later broken.

Reply:
The credentials are created here: https://github.com/aws/aws-toolkit-jetbrains/pull/2693/files/03ead727a93fd234f3179725393c7d26f8f66a19..aff9010a7e697b863fa0d80bce99e503191926fc#diff-48601f6230786dbe8deaf616942766b274bb77efcd36051f856b3d77da011a53R70
I'll add the mock regions and credentials change